rustls/src/crypto/mod.rs

```diff
@@ -256,15 +239,13 @@ impl CryptoProvider {
     /// - installs one `from_crate_features()`, or else
     /// - panics about the need to call [`CryptoProvider::install_default()`]
     pub(crate) fn get_default_or_install_from_crate_features() -> &'static Arc<Self> {
-        if let Some(provider) = Self::get_default() {
-            return provider;
-        }
-
-        let provider = Self::from_crate_features()
-            .expect("no process-level CryptoProvider available -- call CryptoProvider::install_default() before this point");
-        // Ignore the error resulting from us losing a race, and accept the outcome.
-        let _ = provider.install_default();
-        Self::get_default().unwrap()
+        static_default::get_default().unwrap_or_else(|| {
```

Comment:
IMO the previous structure with the `let-else` was easier to follow than the chaining with `unwrap_or_else()` that you have here.

Reply:
reverted (under protest, as I would prefer things to be more functional whenever possible)